code/modules/antagonists/cult/cult_items.dm

```diff
@@ -106,6 +108,129 @@ Striking a noncultist, however, will tear their flesh."}
 		return
 	..()
 
+#define WIELDER_SPELL "wielder_spell"
+#define SWORD_SPELL "sword_spell"
+#define SWORD_PREFIX "sword_prefix"
+
+GLOBAL_LIST_INIT(heretic_paths_to_haunted_sword_abilities,list(
+	// Ash
+	PATH_ASH = list(WIELDER_SPELL = /datum/action/cooldown/spell/jaunt/ethereal_jaunt/ash, \
+	SWORD_SPELL = /datum/action/cooldown/spell/charged/beam/fire_blast, SWORD_PREFIX = "ashen"), \
+	// Flesh
+	PATH_FLESH = list(WIELDER_SPELL = /datum/action/cooldown/spell/pointed/blood_siphon, \
+	SWORD_SPELL = /datum/action/cooldown/spell/pointed/cleave, SWORD_PREFIX = "sanguine"), \
+	// Void
+	PATH_VOID = list(WIELDER_SPELL = /datum/action/cooldown/spell/pointed/void_phase, \
+	SWORD_SPELL = /datum/action/cooldown/spell/cone/staggered/cone_of_cold/void, SWORD_PREFIX = "tenebrous"), \
+	// Blade
+	PATH_BLADE = list(WIELDER_SPELL = /datum/action/cooldown/spell/pointed/projectile/furious_steel/haunted, \
+	SWORD_SPELL = /datum/action/cooldown/spell/pointed/projectile/furious_steel/solo, SWORD_PREFIX = "keen"), \
+	// Rust
+	PATH_RUST = list(WIELDER_SPELL = /datum/action/cooldown/spell/cone/staggered/entropic_plume, \
+	SWORD_SPELL = list(/datum/action/cooldown/spell/aoe/rust_conversion/small, /datum/action/cooldown/spell/pointed/rust_construction), SWORD_PREFIX = "rusted"), \
+	// Cosmic
+	PATH_COSMIC = list(WIELDER_SPELL = /datum/action/cooldown/spell/conjure/cosmic_expansion, \
+	SWORD_SPELL = /datum/action/cooldown/spell/pointed/projectile/star_blast, SWORD_PREFIX = "astral"), \
+	// Lock
+	PATH_LOCK = list(WIELDER_SPELL = /datum/action/cooldown/spell/pointed/burglar_finesse, \
+	SWORD_SPELL = /datum/action/cooldown/spell/pointed/apetra_vulnera, SWORD_PREFIX = "incisive"), \
+	// Moon
+	PATH_MOON = list(WIELDER_SPELL = /datum/action/cooldown/spell/pointed/projectile/moon_parade, \
+	SWORD_SPELL = /datum/action/cooldown/spell/pointed/moon_smile, SWORD_PREFIX = "shimmering"), \
+	// Starter
+	PATH_START = list(WIELDER_SPELL = null, SWORD_SPELL = null, SWORD_PREFIX = "nascent") // lol loser
+))
+
+/obj/item/melee/cultblade/haunted
+	name = "haunted longsword"
+	desc = "An eerie sword with a blade that is less 'black' than it is 'absolute nothingness'. It glows with furious, restrained green energy."
+	icon_state = "hauntedblade"
+	inhand_icon_state = "hauntedblade"
+	worn_icon_state = "hauntedblade"
+	force = 35
+	throwforce = 15
+	block_chance = 55
+	wound_bonus = -25
+	bare_wound_bonus = 30
+	free_use = TRUE
+	light_color = COLOR_BLACK
+	light_system = OVERLAY_LIGHT
+	light_range = 4
+	// holder for the actual action when created.
+	var/datum/action/cooldown/spell/path_wielder_action
+	var/mob/living/trapped_entity
+	var/heretic_path
+
+/obj/item/melee/cultblade/haunted/Initialize(mapload, mob/soul_to_bind, mob/awakener, no_binding)
+	. = ..()
+
+	if(!no_binding || mapload)
```

Comment:
Also it's easier to read if we prevent double negatives in var names
`no_binding = FALSE` -> `do_bind = TRUE`
`if(!no_binding)` -> `if(do_bind)`